extractor/src/main/java/com/example/extractor/ConfigSerializer.java

```diff
@@ -0,0 +1,66 @@
+package com.example.extractor;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser;
+import com.puppycrawl.tools.checkstyle.bdd.TestInputConfiguration;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class ConfigSerializer {
+
+    public static void serializeConfig(String exampleFilePath, String templateFilePath, String outputFilePath) throws Exception {
+        System.out.println("Loading configuration from example file...");
+        TestInputConfiguration testInputConfiguration = InlineConfigParser.parseWithXmlHeader(exampleFilePath);
+        Configuration xmlConfig = testInputConfiguration.getXmlConfiguration();
+
+        String template = new String(Files.readAllBytes(Path.of(templateFilePath)));
+
+        String configContent = replacePlaceholders(template, xmlConfig);
+
+        Files.write(Path.of(outputFilePath), configContent.getBytes());
+    }
+
+    private static String replacePlaceholders(String template, Configuration config) {
+        Configuration targetModule = getFirstRelevantModule(config);
```

Comment:
Please move this method out of this class.
ConfigSerializer should do only serialization to string.
Where such string will be placed, is another story.

Reply:
Done. Moved to a new class called `TemplateProcessor`

Reply:
done. right now, the `TemplateProcessor` just does is replacement.